From d6fa64123cf0b9c6f71915de286237f94de5503e Mon Sep 17 00:00:00 2001
From: b00984 <b00984@freescale.com>
Date: Wed, 6 May 2009 18:11:53 +0800
Subject: [PATCH] ENGR00107628 Android Accelerometer support in imx31 3stack

Adjust poll interval, and set default mode
remove event report in soft irq

Signed-off-by: Shen Yong <b00984@freescale.com>
---
 drivers/hwmon/mxc_mma7450.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/hwmon/mxc_mma7450.c b/drivers/hwmon/mxc_mma7450.c
index 60a2782..1c78631 100644
--- a/drivers/hwmon/mxc_mma7450.c
+++ b/drivers/hwmon/mxc_mma7450.c
@@ -27,7 +27,7 @@
 /*macro define*/
 #define MMA7450_I2C_ADDR	0x1D
 #define DEVICE_NAME		"mma7450"
-#define POLL_INTERVAL		50
+#define POLL_INTERVAL		100
 #define DEBUG
 
 #define INPUT_FUZZ	4
@@ -527,7 +527,6 @@ static void report_abs(void)
 
 static void mma_bh_handler(struct work_struct *work)
 {
-	report_abs();
 }
 
 static void mma7450_dev_poll(struct input_polled_dev *dev)
@@ -715,6 +714,7 @@ static int mma7450_probe(struct i2c_client *client,
 
 	dev_info(&client->dev, "mma7450 device is probed successfully.\n");
 
+	set_mod(1);
 	return 0;		/*what value shall be return */
 
 	/*error handle */
-- 
1.5.4.4

